Release v0.3 into main

AlexanderDolnikov opened the pull request.

AlexanderDolnikov committed:
fixed:
1. tab descriptions
2. rotating loader logo moved a bit lower (it was over the upper elements)
new:
1. dark theme is now default

AlexanderDolnikov committed:
new:
1. full rework of Profile page - user can see profile info (edit not implemented yet)
2. added nice change theme button
3. moved "need to log in" container to another component - with props to use it in different places of the app

to-do:
1. move all content in the app a bit lower, so that it looks ok on all the phones
2. fix log in and sign up pages (visual part) for both themes
3. edit functionality in the profile page

AlexanderDolnikov committed:
design fix for login and sign up pages

AlexanderDolnikov committed:
new:
1. design fix for profile page
2. alert during registration, if user with this email already exists
3. profile page fully working: add, edit text fields and profile photo

AlexanderDolnikov committed:
fixed tabs images and tabs font

mishanya1111 commented on app/_layout.tsx:
```diff
@@ -6,10 +6,38 @@ import { useFonts } from 'expo-font';
+	const { background } = useThemeColors();
+
+	return (
+		<View style={{ flex: 1, backgroundColor: background, paddingTop: 40 }}>
```
если ты используешь background здесь, то проверь можно ли избавиться от компонета BagroundView

AlexanderDolnikov replied:
я не знал про существование BagroundView. исправил так чтобы все настраивалось в BagroundView + с адаптацией под обе платформы. app/_layout.tsx исправил на то, что было изначально

mishanya1111 commented on components/Loader.tsx:
```diff
@@ -39,7 +39,8 @@ const styles = StyleSheet.create({
 		justifyContent: 'center',
-		alignItems: 'center'
+		alignItems: 'center',
+		marginTop: 150
```
Это хреновое решение, ты должен настраивать(изменять расположение лоадера ) компонент ,где ты используешь этот лоадер.

AlexanderDolnikov replied:
добавил значение по умолчанию в Loader, на страницах 1-3 все работает - везде все одинаково и по центру

mishanya1111 commented on components/LoginNeededContainer.tsx:
```diff
@@ -0,0 +1,55 @@
+}
+
+export function LoginNeededContainer({ location }: LoginNeededContainerProps) {
+    const messages: Record<LoginMessageType, string> = {
```
это плохое решение. Что ты будешь делать если в будущем у тебя будут ещё страницы где ты будешь это использовать. Ты будешь каждый раз изменять этот список?
Просто передавай текст пропсом

AlexanderDolnikov replied:
сделал

mishanya1111 commented on components/MeetupDetails.tsx:
```diff
@@ -43,12 +44,8 @@ const MeetupDetails = () => {
-					<Button title="Log In" onPress={() => router.push(SIGN_IN)} />
-				</View>
+
+				<LoginNeededContainer location="meetup_details"/>
```
Изменить компонент LoginNeededContainer  и исправить

mishanya1111 commented on constant/userApi.ts:
Обоснуй вынесению этих методов сюда. (обычно их выносят если они используются несколько раз. Они используются несколько раз?)

mishanya1111 commented on hooks/useSignUp.ts:
```diff
@@ -34,8 +40,13 @@ export const useSignUp = () => {
-			console.log(error.toString());
+
+			if (error.response?.data?.email?.[0] === "user with this email already exists.") {
+				Alert.alert("Error!", "User with this email is already registered, you should login instead.");
```
Alert? не слишком ли жестоко(я не до конца понимаю зачем и как это вообще выглядит)

mishanya1111 replied:
Опять же в чём смысл обрабытывать это ошибку уникально. То есть все остальные ошибки как ошибки а это **царь во дворца** ?

AlexanderDolnikov replied:
исправил на сообщения красным цветом прямо на странице

mishanya1111 commented on app/(tabs)/profile.tsx:
```diff
@@ -1,120 +1,339 @@
+                setPhotoUri(updatedData.photo); // 🔥 Обновляем фото с сервера
+            }
+
+            console.log("Данные профиля успешно обновлены!");
```
вот здесь было бы правильно залить изменения никнейма в AuthContext(конечно если он изменился)

mishanya1111 commented on app/(tabs)/profile.tsx:
```diff
@@ -1,120 +1,339 @@
+                            <Text style={[styles.label, { color: text }]}>Name:</Text>
+                            <TextInput
+                                style={styles.input}
+                                value={editableUserData.first_name || userData?.first_name || ''}
```
Если я правильно понимаю твой код при подгрузке даных они у тебя устанавливаются в editableUserData. Какой смысл проверять наличие userData?.first_name  при отсутствии editableUserData.first_name. я даже скажу больше(хотя вот здесь не факт, уточни у гпт): нет   никакого смысла вот здесь в коце ставить '' так как у тебя при иницализации уже устанавливается значение ''
```
const [editableUserData, setEditableUserData] = useState({
        first_name: '',
        last_name: '',
        username: '',
        email: '',
        user_description: '',
        tg_id: '',
    });
```

mishanya1111 commented on app/(tabs)/profile.tsx:
верни куда-нибудь кнопку 'Create meetup'

mishanya1111 reviewed (changes requested):
Удачи)

AlexanderDolnikov committed:
1. lots of code reorganization and optimization
2. fixed main _layout
3. fixed loader
4. added loader to profile page
5. fixed error messages during registration
6. added "Create meetup" button on "Owned meetups" and "Profile" pages

mishanya1111 commented on components/Loader.tsx:
```diff
@@ -26,7 +29,7 @@ export default function Loader() {
 
 	return (
-		<View style={styles.loader}>
+		<View style={{ ...styles.loader, marginTop: topOffset }}>
```
перепиши
<View style={[styles.loader, { marginTop: topOffset }]}>

mishanya1111 commented on components/LoginNeededContainer.tsx:
```diff
@@ -0,0 +1,45 @@
+        alignItems: 'center',
+        width: '100%',
+        maxWidth: 400,
+        marginTop: 250
```
опять фиксированный отступ 😥, ну да ладно

mishanya1111 commented on hooks/useProfile.ts:
```diff
@@ -0,0 +1,111 @@
+export const getUserData = async (userID: number | null, token: string | undefined) => {
+    try {
+        const response = await axios.get(`${USER_API_URL}${userID}/`, {
+            headers: { Authorization: `Bearer ${token}` }
```
меня это вводит в заблуждение так как токен у нас это объект который имеет поля access и refresh. И вообще у нас так-то есть специальный метод для header
![image](https://github.com/user-attachments/assets/6e8e8621-c633-4200-9ddd-21d81d0cb8fb)

mishanya1111 commented on hooks/useProfile.ts:
```diff
@@ -0,0 +1,111 @@
+    }
+};
+
+export const updateUserData = async (userID: number, token: string, userData: any) => {
```
тоже самое про токен. (это предложение недействительно)также интересно почему ты как я не вынес эти вспомогательные методы в отдельный хук как это я делаю везде чтобы файл разбить на функциональную и визуальную часть

Update1: догортавшись до низа я понял что я ничего не понял. ты где-то из вне используешь ещё раз этот метод(и ещё один в этом файле)?? это просто единственное объяснение почему ты его не внёс в хук

Update2: я разобрался, всё херня, переделывай, напиши это так чтобы это было похоже на мой хук(зайди в любой в этой папке и посмотри)

mishanya1111 commented on hooks/useProfile.ts:
```diff
@@ -0,0 +1,111 @@
+            return;
+        }
+        fetchData();
+    }, [fetchData]);
```
я правильно понимаю что fetchData() только один раз отрабатывает, если да то засунь саму функцию в useEffect а дальше вызывай
```
useEffect(() => {
        function fetchData() { ...}
        if (!userID || !token?.access) {
            setLoading(false);
            return;
        }
        fetchData();
    }, []); ```
    
    update1: можешл оставить стиль 7написание функции, это просто пример

mishanya1111 commented on hooks/useProfile.ts:
хук ~~говно~~ плохой , переделай по примеру моих других хуков

mishanya1111 replied:
опять же, можно написать этот хук так , чтобы использовать его и для просмотра чужого пользователя, тогда он может по стилю чуть отличаться от моего, хотя и не обязательно, просто нкоторый функционал будет заблочен(можно проверять id человека который сейчас в акке и id того на чью страницу ты пришёл )

mishanya1111 reviewed (changes requested):
основная проблема в хуке профиля, сам профиль вроде без проблем(потому что эжто визуал, я не сильно разбирался где какая функция используется)

AlexanderDolnikov committed:
fixed profile with corresponding files

mishanya1111 reviewed (approved):
молодец
